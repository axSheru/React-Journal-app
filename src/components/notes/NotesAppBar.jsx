import React from 'react';

export const NotesAppBar = () => {
    return (
        <div className='notes__appbar'>
            <span>22 de marzo de 2022</span>
            <div>
                <button className='btn'>Picture</button>
                <button className='btn'>Save</button>
            </div>
        </div>
    );
}
