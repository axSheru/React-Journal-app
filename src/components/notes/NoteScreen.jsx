import React from 'react';
import { NotesAppBar } from './NotesAppBar';

export const NoteScreen = () => {
    return (
        <div className='notes__main-content'>
            <NotesAppBar />

            <div className="notes__content">
                <form className='notes__main-content'>
                    <input
                        type="text"
                        placeholder='Some awesome title'
                        className='notes__title-input'
                        autoComplete='off'
                    />
                    <textarea
                        placeholder='What happened today?'
                        className='notes__textarea'
                    ></textarea>
                    <div className="notes__image">
                        <img
                            src="https://i1.wp.com/www.lafosadelrancor.com/wp-content/uploads/2019/07/Sith-Trooper-El-Ascenso-de-Skywalker.jpg?fit=678%2C489&ssl=1"
                            alt="imagen"
                        />
                    </div>
                </form>
            </div>
        </div>
    );
}
