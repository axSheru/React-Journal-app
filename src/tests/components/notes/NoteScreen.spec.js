import configureStore from 'redux-mock-store';
import { mount } from "enzyme";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import { activeNote } from '../../../actions/notes';
import { NoteScreen } from '../../../components/notes/NoteScreen';

jest.mock( '../../../actions/notes', ()  => ({
    activeNote: jest.fn()
}));

    const middlewares = [ thunk ];
    const mockStore = configureStore( middlewares );

    const initState = {
        auth: {
            uid: '1235',
            name: 'Alex'
        },
        notes: {
            active: {
                id: 12345,
                title: 'title',
                body: 'body',
                date: 0
            },
            notes: []
        }
    };

    let store = mockStore( initState );

    store.dispatch = jest.fn();

    const wrapper = mount(
        <Provider store={ store } >
            <NoteScreen />
        </Provider>
    );

describe('Pruebas en el componente NoteScreen.', () => {

    test('Debe de hacer match con el snapshot.', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe de disparar la acciÃ³n activeNote.', () => {

        wrapper.find('input[name="title"]').simulate( 'change', {
            target: {
                name: 'title',
                value: 'Hola de nuevo'
            }
        });

        expect( activeNote ).toHaveBeenLastCalledWith(
            {"body": "body", "date": 0, "id": 12345, "title": "Hola de nuevo"},
            {"body": "body", "date": 0, "id": 12345, "title": "Hola de nuevo"}
        );
        
    });

});