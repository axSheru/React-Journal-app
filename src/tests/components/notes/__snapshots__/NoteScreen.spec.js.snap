// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente NoteScreen. Debe de hacer match con el snapshot. 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      22 de marzo de 2022
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <form
      className="notes__main-content"
    >
      <input
        autoComplete="off"
        className="notes__title-input"
        name="title"
        onChange={[Function]}
        placeholder="Some awesome title"
        type="text"
        value="title"
      />
      <textarea
        className="notes__textarea"
        name="body"
        onChange={[Function]}
        placeholder="What happened today?"
        value="body"
      />
    </form>
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
